src/app/core/entity-components/entity-list/entity-list.component.ts

```diff
@@ -109,10 +112,9 @@ export class EntityListComponent<T extends Entity>
       this.displayColumnGroup(this.defaultColumnGroup);
     }
     if (changes.hasOwnProperty("allEntities")) {
-      this.filteredEntities = this.allEntities;
-      this.initFilterSelections();
+      await this.initFilterSelections();
+      this.applyFilterSelections();
     }
-    this.loadUrlParams();
```

Comment:
There is a slight problem with this change. If you visit the children list and don't click anything the schools are never shown but only once a different view is selected. This line was responsible for reloading the view once the schools are fully loaded. This also allows to load the component with a schools filter selected in the params.